### PR TITLE
Add prompts to user to create metadata YAML file

### PR DESCRIPTION
**Pull request by kkoetje**
User is prompted at end of workflow to create metadata YAML file. Functions that create YAMLs live in their own file mission_yaml_files.py which has been added to the workflow. Mods to yaml structure and wording can be made there.

### CODE REVIEW

**Review by SBFRF — CHANGES_REQUESTED** [review]

File: mission_yaml_files.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+    # Define the preamble text
+    preamble = "# This is a YAML file containing a dictionary of user responses to the following questions about the mission:\n" + "\n".join(f"#     {item}" for item in questions) + "\n"
+
+    # Create yaml file and write dictionary to it
+    yfname = r"mission_summary_metadata.yaml"
+
+    # Create the full file path
+    file_path = os.path.join(datadir, yfname)
+
```
[the remark this answers is not in the document]
Comment (kkoetje) [reply mission_yaml_files.py:66]: Resolved at beginning of each function.

**Comment by SBFRF** [comment]
Running through example with Nick d now- comments:
- Default behavior is making a pos file ... I thought we turned that to default look for a pos file
- Ran 20240930 and ran into a timestamp error ... might be associated with making `pos` file
- file mission_summary_metadata.yaml check came up as already existing, questioned overwrite.
- mechanical failure type suggested \n between type
- hitting o (instead of 0) created code to bomb on mechanical failure yaml creation, suggest asking for input again with error regurgitated to user (e.g. didn't recognize that input-try again)
- for mission failure yaml, we keep getting errors requiring an integer, tried 1/0

**Comment by kkoetje** [comment]
> Running through example with Nick d now- comments:
> 
> * Default behavior is making a pos file ... I thought we turned that to default look for a pos file
> * Ran 20240930 and ran into a timestamp error ... might be associated with making `pos` file
> * file mission_summary_metadata.yaml check came up as already existing, questioned overwrite.
> * mechanical failure type suggested \n between type
> * hitting o (instead of 0) created code to bomb on mechanical failure yaml creation, suggest asking for input again with error regurgitated to user (e.g. didn't recognize that input-try again)
> * for mission failure yaml, we keep getting errors requiring an integer, tried 1/0

**Comment by kkoetje** [comment]
> > Running through example with Nick d now- comments:
> > 
> > * Default behavior is making a pos file ... I thought we turned that to default look for a pos file
> > * Ran 20240930 and ran into a timestamp error ... might be associated with making `pos` file
> > * file mission_summary_metadata.yaml check came up as already existing, questioned overwrite.
> > * mechanical failure type suggested \n between type
> > * hitting o (instead of 0) created code to bomb on mechanical failure yaml creation, suggest asking for input again with error regurgitated to user (e.g. didn't recognize that input-try again)
> > * for mission failure yaml, we keep getting errors requiring an integer, tried 1/0

All feedback above has been addressed -- significant improvement to error handling have been made, should be much more stable now.

File: mission_yaml_files.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
+# Contains function to make a mission summary YAML and a function to make a mission failure report YAML (in progress)
+
+# Mission Summary YAML File
+def make_summary_yaml(datadir):
+
+    import yaml
```
[the remark this answers is not in the document]
Comment (kkoetje) [reply mission_yaml_files.py]: Import statements are now global. Issue created for requirements.txt.

**Comment by SBFRF** [comment]
Awesome, is this ready for merge?

**Comment by kkoetje** [comment]
Yes!

**Review by SBFRF — APPROVED** [review]
looks great, thanks @kkoetje

**Pull request merged**